### PR TITLE
Cookie Consent: backend changes to move cookie consent block to template part

### PR DESCRIPTION
**Pull request by arcangelini**
## Proposed changes:

This echos the cookie consent content into the footer if the template part exists.

More to come...

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to '..'
*

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/cookie-consent/cookie-consent.php [hunk 1]
```diff
@@ -85,6 +85,37 @@ function load_assets( $attr, $content ) {
 	);
 }
 
+/**
+ * Render the cookie consent template-part.
+ *
+ * @since 12.0
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/cookie-consent/cookie-consent.php]:
```suggestion
 * @since $$next-version$$
```

**Review by jeherve — COMMENTED** [review]
Is there a discussion about this? I'd be curious to learn more about the change!

**Comment by arcangelini** [comment]
> Is there a discussion about this? I'd be curious to learn more about the change!

@jeherve we are on our meetup and are working live to make the Cookie Consent a template part rather than a block. This was based of some talks in a P2 - pdDR7T-Cr-p2#comment-803.

There is a front end PR that is soon to come as well. Also, do you have any thoughts on:

```
echo do_blocks( $template_part->content ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

We had to ignore the linter to get the PR started, but I think maybe this isn't the best option? The problem is if we escape it the template breaks when inserting.

**Comment by jeherve** [comment]
> I think maybe this isn't the best option? The problem is if we escape it the template breaks when inserting.

Do we escape the contents of the template part earlier on our end? If so, that may be okay / enough.

File: projects/plugins/jetpack/extensions/blocks/cookie-consent/cookie-consent.php [hunk 2]
```diff
@@ -85,6 +85,37 @@ function load_assets( $attr, $content ) {
...
+ */
+function render_cookie_consent_template() {
+
+	if ( is_admin() ) {
+		return;
+	}
+
+	$template_part_name = 'cookie-consent';
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/cookie-consent/cookie-consent.php]: Please rename this to `cookie-consent-block-template-part`

**Comment by alshakero** [comment]
Since [the other PR is ready](https://github.com/Automattic/jetpack/pull/30903). Can we get his cleaned up and merged in that PR?

**Review by alshakero — CHANGES_REQUESTED** [review]

**Comment by alshakero** [comment]
**Note**: This was merged into https://github.com/Automattic/jetpack/pull/30903.

**Pull request merged**